### DISCUSSION TITLE
Is there a way to retrieve the visible range of the current Y axis?

### DISCUSSION THREAD

**Discussion by deltamelter** [dbuezas/lovelace-plotly-graph-card#504] [Q&A]
Mainly looking at methods to adjust Y zoom so that the Y axis range is limited to what is actually within visibility for the current X range..
For example, there was an extreme high several days ago, but the Y-axis range auto scaling keeps the Y-Range to include that high, leaving more than half of the chart empty when looking at just today.
default view of the chart is forward looking (the y-range looking at everything past and future):
```
time_offset: 34h
hours_to_show: 36h
```
It would also be useful when drawing time related shapes and variable width bars representing timed events to set the "height" to a reasonable value within the currently visible range. instead of simply blocking out y: 0 to some arbitrary number that might be too big or small to see in context of the other data in the same Yaxis.

**Comment by dbuezas** [comment]
Yes, you can get the visible x range with `get('visible_range')` and then use that to filter values and grab min and max. Something like this:

```yaml
entities:
   - entity : sensor.xyz
     filters:
       - filter: |
            x >= get('visible_range')[0] && x <= get('visible_range')[1]
     fn: |
        $ex {
           vars.y_range = [
                 ys.reduce((y, acc)=> Math.min(y, acc),Number.POSITIVE_INFINITY),
                 ys.reduce((y, acc)=> Math.max(y, acc),Number.NEGATIVE_INFINITY)
           ]
         }
```

**Reply by deltamelter** [reply]
I only got errors saying 
```
Error: at [entities.0.filters.0]: Error in filter: ReferenceError: get is not defined
```

**Reply by dbuezas** [reply]
```yaml
entities:
   - entity : sensor.xyz
     fn2: $ex vars.vr=get('visible_range')
     filters:
       - filter: |
            x >= vars.vr[0] && x <= vars vr[1]
```
